### ISSUE TITLE
Activities not working as expected

### ISSUE THREAD

**Issue by simonrolph** [simonrolph/NatureCounts#7]
Producing the error

Signed in as s.rolph@wildsheffield.com > Activities > create new activity

Selected: The group is closed and membership is by invite only (although I need some clarity on how these 2 options work)

Set where to wyming brook  (SRWT reserve)

![image](https://user-images.githubusercontent.com/17750766/100730830-7b493780-33c2-11eb-8c16-843325b9e630.png)

Created new activity:
![image](https://user-images.githubusercontent.com/17750766/100731009-bc414c00-33c2-11eb-9dba-337dfca7663a.png)

Sending the link: https://record.wildsheffield.com/join/test-activity-2 to my other account (simon.p.rolph@gmail.com) just gives page not found:
![image](https://user-images.githubusercontent.com/17750766/100731153-ed218100-33c2-11eb-8518-99796739f817.png)

Sending invite by email works and the email is received but following the link gives me this: 
![image](https://user-images.githubusercontent.com/17750766/100731303-1c37f280-33c3-11eb-818b-29bd8d29fdfe.png)

On browse all activities it says I am an admin:
![image](https://user-images.githubusercontent.com/17750766/100731430-47224680-33c3-11eb-9623-abec59f5cba9.png)

But on my activities it only shows one member:
![image](https://user-images.githubusercontent.com/17750766/100731488-5d300700-33c3-11eb-9a22-facf0e4af8fe.png)

![image](https://user-images.githubusercontent.com/17750766/100731548-7042d700-33c3-11eb-80f4-cb7e5d0c0e73.png)

Note that the simon.p.rolph@gmail account has the first name Simon and last name Test (so I can differentiate the 2), but this account doesn't appear in the list.

Problems are: 

 * the 'send a link' invite option doesn't work
 * inviting by email gives you access to the page as an admin but doesn't show you in the list of members

**Comment by johnvanbreda** [comment]
1. I've fixed the send a link invite option - this was copied from iRecord and I hadn't ported it to D8. Note though, the join link icon should not be available for invite only groups as that's intended for wider sharing (e.g. social media). The specific invite functionality is intended for invite only groups. So now this icon won't appear for the groups you've created.
2. I think the confusion here is that although you sent the link to your other email account, when you followed the link you were already logged in as s.rolph@wildsheffield.com. To do this test correctly you need to log out before following the link, or log out then log in again as your other account.

**Comment by simonrolph** [comment]
How do you delete activities?

I had bad moment and I accidentally deleted the 'add/edit and activity' page because I was looking for the delete button... I've added a new one which I seems to work fine, could you check the options to see if there's anything I've missed in the set-up... thanks (and sorry)

**Comment by johnvanbreda** [comment]
If you edit the page, then add the "Include date fields" option you can set an end-date for activities to "expire" them. That's logically simpler than an actual delete where you may have to remove the activity from records etc.

Looking at the options, it's up to you how it's configured. Here are a few suggestions:
* You might like to think about whether the "Include members control" option should be checked, as this option doesn't work so well for long member lists. You can view the members in other places anyway.
* Include private records field is ticked - this allows groups to be setup which don't release their records. Obviously we don't want this to happen without good reason.
* Have a think about which group types are useful in this context.

**Comment by simonrolph** [comment]
Thanks. I accidentally created duplicate activities because every activity that I created as a 'Project' group type it disappeared from the all activities list and the only way I could access it was through the URL trying out different id values. Any idea why that happens?

https://record.wildsheffield.com/activities/edit?group_id=1843

**Comment by johnvanbreda** [comment]
Yes, the activities list page has a filter on the group types. On the edit tab you'll find this in the preset parameter values:
```
group_type_id=4842,4843,4844,4845,5879,6151
```

Here's the list of group types defined on the warehouse (excluding some project specific ones):

3252 | Recording group
3253 | Project
3254 | Organisation
4842 | Events & bioblitzes
4843 | Recording forms
4844 | National groups
4845 | Local groups

So, you need to make sure the group types on your edit activity page match the filter on the list page.

**Comment by simonrolph** [comment]
Ok thanks, makes sense

I think the reason I was having invite to activity trouble is my main nature counts account was created with the email simon.p.rolph@gmail.com but then changed to my work email later. So when I created a second account with simon.p.rolph@gmail.com for testing for invitations it's actually thinks they're the same profile - even show the same records on the 'my records page'. I'll close this since I think I've got my head around activities.
